iwyu_ast_util.cc

```diff
@@ -1310,6 +1310,12 @@ bool IsImplicitlyInstantiatedDfn(const FunctionDecl* decl) {
              clang::TSK_ImplicitInstantiation;
 }
 
+const CXXMethodDecl* GetUpmostInHierarchy(const CXXMethodDecl* decl) {
+  while (decl->size_overridden_methods())
```

Comment:
I'm not sure it's cleaner, but isn't the least derived/most basic method `std::prev(decl->end_overridden_methods())`? Maybe that iterator is not backward-friendly, though.

Reply:
I tried that and it doesn't work. I suspect that `overridden_method_range` includes only method declarations from one level, i.e. there is more than one item only in the case of a diamond hierarchy with virtual inheritance.

Reply:
```cpp
      // A method can only override one function from each base class. We
      // don't track indirectly overridden methods from bases of bases.
```
https://github.com/llvm/llvm-project/blob/llvmorg-20-init/clang/lib/Sema/SemaDecl.cpp#L8847

Reply:
Got it, thanks